c7n/filters/core.py

```diff
@@ -578,7 +612,10 @@ def process_value_type(self, sentinel, value, resource):
             # comparisons is intuitively wrong.
             return value, sentinel
         elif self.vtype == 'cidr':
-            s = parse_cidr(sentinel)
+            if isinstance(sentinel, list):
```

Comment:
the cyclomatic complexity on value filter is pretty high already, instead of adding a nested conditional, could you keep it flat ie `self.vtype == 'cidr'` and `self.op == 'cidr_overlap'` followed by `self.vtype == 'cidr`